packages/autocertifier/src/Database.ts

```diff
@@ -0,0 +1,160 @@
+    }
+
+    public async start(): Promise<void> {
+        this.db = await open({
+            filename: this.databaseFilePath,
+            driver: sqlite3.Database
+        })
+
+        const result = await this.db.get("SELECT name FROM sqlite_master WHERE type='table' AND name=?", "subdomains")
+
+        if (!result) {
+            await this.createTables()
+        }
+
+        this.createSubdomainStatement = await this.db.prepare("INSERT INTO subdomains (subdomainName, ip, port, token) VALUES (?, ?, ?, ?)")
+        this.getSubdomainStatement = await this.db.prepare("SELECT * FROM subdomains WHERE subdomainName = ?")
+        this.getSubdomainWithTokenStatement = await this.db.prepare("SELECT * FROM subdomains WHERE subdomainName = ? AND token = ?")
+        this.updateSubdomainIpAndPortStatement = await this.db.prepare("UPDATE subdomains SET ip = ?, port = ? WHERE subdomainName = ? AND token = ?")
+        this.getSubdomainAcmeChallengeStatement = await this.db.prepare("SELECT acmeChallenge FROM subdomains WHERE subdomainName = ?")
+        this.updateSubdomainAcmeChallengeStatement = await this.db.prepare("UPDATE subdomains SET acmeChallenge = ? WHERE subdomainName = ?")
+
+        logger.info('Database is running')
+    }
+
+    public async stop(): Promise<void> {
+        if (this.createSubdomainStatement) {
+            await this.createSubdomainStatement.finalize()
+        }
+        if (this.getSubdomainStatement) {
+            await this.getSubdomainStatement.finalize()
+        }
+        if (this.getSubdomainWithTokenStatement) {
+            await this.getSubdomainWithTokenStatement.finalize()
+        }
+        if (this.updateSubdomainIpAndPortStatement) {
+            await this.updateSubdomainIpAndPortStatement.finalize()
+        }
+        if (this.getSubdomainAcmeChallengeStatement) {
+            await this.getSubdomainAcmeChallengeStatement.finalize()
+        }
+        if (this.updateSubdomainAcmeChallengeStatement) {
+            await this.updateSubdomainAcmeChallengeStatement.finalize()
+        }
+        if (this.db) {
+            await this.db.close()
+        }
+    }
+
+    private async createTables(): Promise<void> {
+        const query = `
+            BEGIN TRANSACTION;
+            CREATE TABLE subdomains (
+                id INTEGER PRIMARY KEY,
+                subdomainName TEXT NOT NULL,
+                ip TEXT NOT NULL,
+                port TEXT NOT NULL,
+                token TEXT NOT NULL,
+                acmeChallenge TEXT,
+                createdAt DATETIME DEFAULT CURRENT_TIMESTAMP
+            );
+            CREATE INDEX subdomain_index on subdomains(subdomainName);
+            COMMIT;
+        `
+        await this.db!.exec(query)
+    }
+}
+
+export interface Subdomain {
+    // TODO: could be called just name / label?
+    subdomainName: string
+    ip: string
+    port: string
+    token: string
+    acmeChallenge?: string | null
```

Comment:
Should use one value for no-value-exists type (`undefined` or `null`)

```suggestion
    acmeChallenge?: string
```